core/primitives/src/views.rs

```diff
@@ -345,12 +346,39 @@ pub struct DebugBlockStatus {
     pub timestamp_delta: u64,
 }
 
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct PeerInfoView {
+    pub addr: Option<SocketAddr>,
+    pub account_id: Option<AccountId>,
+    pub height: BlockHeight,
+    pub tracked_shards: Vec<ShardId>,
+    pub archival: bool,
+}
+
+#[cfg_attr(feature = "deepsize_feature", derive(deepsize::DeepSizeOf))]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct NetworkInfoView {
+    pub peer_max_count: u32,
+    pub num_connected_peers: usize,
+    pub connected_peers: Vec<PeerInfoView>,
+    pub sent_bytes_per_sec: u64,
+    pub received_bytes_per_sec: u64,
+    pub peer_counter: usize,
+}
+
 #[cfg_attr(feature = "deepsize_feature", derive(deepsize::DeepSizeOf))]
 #[derive(Serialize, Deserialize, Debug)]
 pub struct DetailedDebugStatus {
     pub last_blocks: Vec<DebugBlockStatus>,
 }
 
+#[cfg_attr(feature = "deepsize_feature", derive(deepsize::DeepSizeOf))]
```

Comment:
You don't need the DeepSizeOf decorator here

Reply:
I realized that one of its parent struct (`DetailedDebugStatus`) needs this trait, so I am adding it back.